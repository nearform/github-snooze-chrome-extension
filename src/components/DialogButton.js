import * as React from 'react'
import {
  Button,
  IconButton,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle
} from '@mui/material'

export default function DialogButton({
  label,
  title,
  description,
  onConfirm,
  icon = null
}) {
  const [open, setOpen] = React.useState(false)

  const handleClickOpen = () => {
    setOpen(true)
  }

  const handleConfirm = () => {
    setOpen(false)
    onConfirm()
  }

  const handleClose = () => {
    setOpen(false)
  }

  const renderButton = () => {
    if (icon) {
      return <IconButton onClick={handleClickOpen}>{icon}</IconButton>
    }
    return (
      <Button
        fullWidth
        variant="outlined"
        color="secondary"
        onClick={handleClickOpen}
      >
        {label}
      </Button>
    )
  }

  return (
    <div>
      {renderButton()}
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            {description}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleClose}>Cancel</Button>
          <Button onClick={handleConfirm} autoFocus>
            Confirm
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  )
}
