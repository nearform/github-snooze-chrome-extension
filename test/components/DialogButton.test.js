import React from 'react'
import renderer from 'react-test-renderer'
import DialogButton from '../../src/components/DialogButton'

const props = {
  label: 'label',
  title: 'title',
  description: 'description',
  onConfirm: jest.fn()
}

describe('DialogButton.js', () => {
  test('shows the proper DialogButton', () => {
    const tree = renderer.create(<DialogButton {...props} />)

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
