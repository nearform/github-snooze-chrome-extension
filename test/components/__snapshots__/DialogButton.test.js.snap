// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogButton.js shows the proper DialogButton 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-fullWidth css-slst40-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    label
  </button>
</div>
`;
